python/datasources/acs_poverty.py

```diff
@@ -311,80 +321,164 @@ def accumulate_acs_data(self, data):
     # Splits the in memory aggregation into dataframes
 
     def split_data_frames(self):
-        state_data = []
-        county_data = []
+        race_state_data = []
+        sex_state_data = []
+        age_state_data = []
+
+        race_county_data = []
+        sex_county_data = []
+        age_county_data = []
 
         # Extract keys from self.data Tuple
-        # (state_fip, County_fip, Age, Sex, Race): {PopulationObj}
+        # (state_fip, county_fip, Age, Sex, Race): {PopulationObj}
         for data, population in self.data.items():
             state_fip, county_fip, age, sex, race = data
 
             population = self.data[data]
-            above = population[PovertyPopulation.ABOVE]
             below = population[PovertyPopulation.BELOW]
 
+            if (age is not None) + (sex is not None) + (race is not None) != 1:
+                raise AssertionError(f"Invalid Tuple: {data}")
+
             if county_fip is None:
-                state_data.append(
-                    [
-                        state_fip,
-                        self.state_fips[state_fip],
-                        race,
-                        age,
-                        sex,
-                        above,
-                        below,
-                    ]
-                )
+                if race:
+                    race_state_data.append(
+                        [
+                            state_fip,
+                            self.state_fips[state_fip],
+                            race,
+                            below,
+                        ]
+                    )
+                elif sex:
+                    sex_state_data.append(
+                        [
+                            state_fip,
+                            self.state_fips[state_fip],
+                            sex,
+                            below,
+                        ]
+                    )
+                elif age:
+                    age_state_data.append(
+                        [
+                            state_fip,
+                            self.state_fips[state_fip],
+                            age,
+                            below,
+                        ]
+                    )
             else:
-                county_data.append(
-                    [
-                        state_fip,
-                        self.state_fips[state_fip],
-                        state_fip + county_fip,
-                        self.county_fips[(state_fip, county_fip)],
-                        race,
-                        age,
-                        sex,
-                        above,
-                        below,
-                    ]
-                )
+                if race:
+                    race_county_data.append(
+                        [
+                            state_fip,
```

Comment:
Most of the array you append is basically the same for all of the clauses.. can you pull it out into a variable above and just add to that, e.g. race_county_data.append(county_base + [race, below]) or the like? Same for the state data in the section above

Reply:
Done